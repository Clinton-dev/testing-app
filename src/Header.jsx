import React from 'react'

function Header({ text }) {
    return (
        <header>
            <h3 data-testid="h1tag" className="fancy-h1">{text}</h3>
        </header>
    )
}

export default Header