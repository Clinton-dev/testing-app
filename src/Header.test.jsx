import React from "react";
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom"
import Header from './Header'

afterEach(cleanup)
it("renders", () => {
    const { asFragment } = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
})

it("inserts text in h1", () => {
    const { getByTestId, getByText } = render(<Header text="Hello!" />)
    expect(getByTestId('h1tag')).toHaveTextContent('Hello!')
    expect(getByText("Hello!")).toHaveClass("fancy-h1")
})