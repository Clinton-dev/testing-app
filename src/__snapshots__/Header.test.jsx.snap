// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h3
      class="fancy-h1"
      data-testid="h1tag"
    />
  </header>
</DocumentFragment>
`;
